test/gleam/string_test.gleam

```diff
@@ -364,3 +364,69 @@ pub fn to_option_test() {
   |> string.to_option
   |> should.equal(Some("ok"))
 }
+
+pub fn first_test() {
+  ""
+  |> string.first
+  |> should.be_error
+
+  "gleam"
+  |> string.first
+  |> should.equal(Ok("g"))
+
+  "⭐️ Gleam"
+  |> string.first
+  |> should.equal(Ok("⭐️"))
+}
+
+pub fn last_test() {
```

Comment:
Could we have a test for a single letter string such as `"a"` please?

Reply:
Good catch! I've added them now and fixed the pattern matching to include single-letter strings.